packages/runtime/src/runtime.ts

```diff
@@ -225,29 +218,35 @@ export class Runtime {
     this.ctx.gtxs = [encTx];
   }
 
-  // creates new application and returns application id
+  /**
+   * creates new application and returns application id
+   * @param flags SSCDeployment flags
+   * @param payFlags Transaction parameters
+   * @param program approval program
+   */
   async addApp (flags: SSCDeploymentFlags, payFlags: TxParams, program: string): Promise<number> {
     const sender = flags.sender;
     const senderAcc = this.assertAccountDefined(this.store.accounts.get(sender.addr));
 
     // create app with id = 0 in globalApps for teal execution
     const app = senderAcc.addApp(0, flags);
-    this.ctx.state.globalApps.set(app.id, app.attributes);
+    this.ctx.state.accounts.set(senderAcc.address, senderAcc);
+    this.ctx.state.globalApps.set(app.id, senderAcc.address);
 
     this.makeAndSetCtxAppCreateTxn(flags, payFlags);
     await this.run(program); // execute TEAL code with appId = 0
 
     // create new application in globalApps map
-    this.store.globalApps.set(++this.appCounter, app.attributes);
+    this.store.globalApps.set(++this.appCounter, senderAcc.address);
 
+    const attributes = this.assertAppDefined(senderAcc.createdApps.get(0), 0);
     this.ctx.state.globalApps.delete(0); // remove zero app from context after execution
     senderAcc.createdApps.delete(0); // remove zero app from sender's account
 
-    // set new application in sender's account
-    // after setting in globalApps
-    // TODO: will be updated in https://www.pivotaltracker.com/story/show/176487715
-    senderAcc.addApp(this.appCounter, flags);
+    senderAcc.createdApps.set(this.appCounter, attributes);
     this.store.accounts.set(sender.addr, senderAcc);
+    this.ctx.state.accounts.set(senderAcc.address, senderAcc);
```

Comment:
Same as above. Also, do we need this line (keeping `ctx` after transaction is executed) ? I mean `ctx` will be overwritten on `runtime.executeTx` and in `Runtime` we are using `store` to getAccount.

Reply:
same as above, yes, we don't need it.